### PR TITLE
HHH-6951 IdentifiableType.getIdType() should respect @IdClass

### PR DESCRIPTION
**Pull request by quaff**
Before this commit, `IdentifiableType.getIdType()` returns unexpected `null` if multiple @Id present.

[Please describe here what your change is about]

----------------------
By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt)
and can be relicensed under the terms of the [LGPL v2.1 license](https://www.gnu.org/licenses/old-licenses/lgpl-2.1.txt) in the future at the maintainers' discretion.
For more information on licensing, please check [here](https://github.com/hibernate/hibernate-orm/blob/main/CONTRIBUTING.md#legal).

----------------------

https://hibernate.atlassian.net/browse/HHH-6951

### CODE REVIEW

File: hibernate-core/src/test/java/org/hibernate/orm/test/jpa/metamodel/MixedIdAndIdClassHandling.java [hunk 1]
```diff
@@ -50,8 +54,16 @@ public void testAccess(EntityManagerFactoryScope scope) {
 		assertEquals( 1, entityType.getSupertype().getIdClassAttributes().size() );
 
 		assertFalse( entityType.hasSingleIdAttribute() );
+	}
 
-		assertEquals( String.class, entityType.getIdType().getJavaType() );
```
Comment (gavinking) [inline hibernate-core/src/test/java/org/hibernate/orm/test/jpa/metamodel/MixedIdAndIdClassHandling.java:54]: Why did you delete this assertion? Was that an accident?

**Review by gavinking — COMMENTED** [review]
Thanks. This LGTM, apart from the one comment about: from the diff it looks like an assertion got removed... am I misreading the diff?

File: hibernate-core/src/test/java/org/hibernate/orm/test/jpa/metamodel/MixedIdAndIdClassHandling.java (hunk 1 above)
Comment (quaff) [reply hibernate-core/src/test/java/org/hibernate/orm/test/jpa/metamodel/MixedIdAndIdClassHandling.java:54]: No, `entityType.getIdType().getJavaType()` returns `EmployeeId.class` now, this behavior is covered by another dedicated test.

Comment (quaff) [reply hibernate-core/src/test/java/org/hibernate/orm/test/jpa/metamodel/MixedIdAndIdClassHandling.java:54]: Sorry, `entityType.getIdType().getJavaType()` still returns `String.class` for back compatibility, but the assertion is moved to dedicated  test `testGetIdType()`.

Comment (gavinking) [reply hibernate-core/src/test/java/org/hibernate/orm/test/jpa/metamodel/MixedIdAndIdClassHandling.java:54]: I see now.

**Comment by cbsingh1** [comment]
I am still encountering this issue in Spring Boot 3.5.7, which uses hibernate-core 6.6.33.Final.

According to the release history, version 6.6.33.Final was released on October 10, 2025, while the fix in question was merged in August 2024 (see the attached snippet for details).

This may be a silly question, but how can I verify which Hibernate release includes this fix?

image

**Comment by quaff** [comment]
It's available in 7.x, not backported to 6.x.

**Comment by beikov** [comment]
The Jira ticket [HHH-6951](https://hibernate.atlassian.net/browse/HHH-6951) has a field "Fix versions" where the versions are listed.

[HHH-6951]: https://hibernate.atlassian.net/browse/HHH-6951?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
